File: lua/010/main.lua
--[[
    ProjectEuler 010
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local n = 2000000
    local r = math.floor(math.sqrt(n))
    local table = {}
    table[1] = false
    for i = 2, n - 1 do
        table[i] = true
    end
    for i = 2, r do
        if table[i] then
            for j = i * i, n - 1, i do
                table[j] = false
            end
        end
    end
    local sum = 0
    for i = 1, n - 1 do
        if table[i] then
            sum = sum + i
        end
    end
    print(sum)
end

main()

File: julia/010/main.jl
# ProjectEuler 010
# Julia / main.jl
# copyright (c) 2015 Susisu
#
# Julia v0.4.1
# `julia main.jl`

function main()
    n = 2000000
    table = trues(n - 1)
    table[1] = false
    r = Int(floor(sqrt(n)))
    for i = 2:r
        if table[i]
            for j = (i * i):i:(n - 1)
                table[j] = false
            end
        end
    end

    sum = 0
    for i = 1:(n - 1)
        if table[i]
            sum += i
        end
    end
    println(sum)
end

main()
